functions/owasp/check_security.go

```diff
@@ -101,8 +101,7 @@ func (cd CheckSecurity) RunRule(nodes []*yaml.Node, context model.RuleFunctionCo
 
 				}
 
-				if len(opValue.Security) <= 0 && globalSecurity != nil &&
-					(globalSecurity[0].Value.Requirements == nil || globalSecurity[0].Value.Requirements.Len() <= 0) {
+				if opValue.Security != nil && len(opValue.Security) <= 0 {
```

Comment:
Is there a reason you removed the check for global security here?

Reply:
Oh I think I see it, I'm missing a case for a spec like this?:

```yaml
openapi: 3.0.1
info:
  title: Test spec
  version: 0.0.1
security:
  - {}
paths:
  /insecure:
    get:
      responses:
        default:
          description: Default response
```

Reply:
I've added a test case where the second entry in the global security is an empty object, making security optional. I think that's what the `(globalSecurity[0].Value.Requirements == nil || globalSecurity[0].Value.Requirements.Len() <= 0)` was doing, but it was only checking the 0th element?